### PR TITLE
resource_manager: add a new runaway action SwitchGroup

### PR DESCRIPTION
**Pull request by JmPotato**
Part of https://github.com/pingcap/tidb/issues/54434.

### CODE REVIEW

File: proto/resource_manager.proto [hunk 1]
```diff
@@ -219,8 +220,11 @@ message RunawayWatch {
 }
 
 message RunawaySettings {
-  RunawayRule rule = 1; 
+  RunawayRule rule = 1;
   RunawayAction action = 2;
+  // When the runaway action is `SwitchGroup`,
+  // this field will be used to indicate which group to switch.
+  string switch_group_name = 4;
```
Comment (nolouch) [inline proto/resource_manager.proto:227]: move this line `below watch=3`.

**Review by nolouch — APPROVED** [review]
rest lgtm

**Review by HuSharp — APPROVED** [review]

**Comment by nolouch** [comment]
ptal @overvenus

**Review by overvenus — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**